packages/cli/src/scaling/job-processor.ts

```diff
@@ -116,13 +123,20 @@ export class JobProcessor {
 			executionTimeoutTimestamp,
 		);
 
+		const { pushRef } = job.data;
+
 		additionalData.hooks = WorkflowExecuteAdditionalData.getWorkflowHooksWorkerExecuter(
 			execution.mode,
 			job.data.executionId,
 			execution.workflowData,
-			{ retryOf: execution.retryOf as string },
+			{ retryOf: execution.retryOf as string, pushRef },
 		);
 
+		if (pushRef) {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+			additionalData.sendDataToUI = WorkflowExecuteAdditionalData.sendDataToUI.bind({ pushRef });
```

Comment:
Could we do this somewhere where the additionalData is instantiated instead of overwriting it like this?

Reply:
I'm following how it's being done for regular mode:

https://github.com/n8n-io/n8n/blob/c6d97ec24196182bef9af9126a9cd8761b2b9113/packages/cli/src/workflow-runner.ts#L278-L280

Looks like additional data has always been handled like this - notice also `setExecutionStatus` is being overwritten. I'd prefer to follow what exists and once we simplify additional data (which we definitely should) then we can simplify it everywhere rather than having queue mode and regular mode differ on this.